src/app/shared/components/grid-list/grid-list.component.ts

```diff
@@ -24,6 +25,18 @@ export class GridListComponent implements OnInit {
         return `item-span-${ this.itemColumnSpan }`;
     }
 
+    get gridGapClass(): string {
+        const baseString = "grid-gap-";
+        const validGapValues = [0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8, 9, 10,
+            11, 12, 14, 16, 20, 24, 28, 32, 36, 40, 44, 48, 52, 56, 60, 64, 72, 80, 96];
+
+        if (validGapValues.includes(this.gridGap)) {
+            return baseString + this.gridGap.toString().replace(".", "-");
+        }
+
+        return baseString + "0";
+    }
```

Comment:
this is a bit of an overkill. We are only applying classes here, so any invalid values will not throw errors or bugs it simply will not apply any styles

simple string interpolation will do

Reply:
done